### PR TITLE
ocamlPackages.lwt-canceler: init at 0.2

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Motivation for this change

This is a continuation of PR #129444, #129509, and #129511 which are prerequisites to make LIGO build under nix.
The version is the specific version that LIGO needs that's why it's not the latest on opam.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/lwt-canceler/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildDunePackage, lwt }:
+
+buildDunePackage {
+  pname = "lwt-canceler";
+  version = "0.2";
+
+  src = builtins.fetchurl {
+    url = https://gitlab.com/nomadic-labs/lwt-canceler/-/archive/v0.2/lwt-canceler-v0.2.tar.gz;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/lwt-canceler/default.nix:8]: What about using `fetchFromGitLab`?

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/lwt-canceler/default.nix:8]: Did not know it existed, these archives are what opam points to. But if you rather I use `fetchFromGitLab` I can change all my PRs to use that

Comment (vbgl) [reply pkgs/development/ocaml-modules/lwt-canceler/default.nix:8]: Yes, it is better to use fetchFromGitLab (or fetchzip); see the relevant documentation: https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/lwt-canceler/default.nix]: I have now updated this PR and will continue with the others, thank you

**Pull request merged**